public/js/components/Editor.js

```diff
@@ -141,6 +142,22 @@ const Editor = React.createClass({
     this.editor.setText(text);
   },
 
+  setMode(sourceText) {
+    const contentType = sourceText.get("contentType");
+
+    if (contentType.includes("javascript")) {
+      this.editor.setMode({ name: "javascript" });
+    } else if (contentType === "text/wasm") {
+      this.editor.setMode({ name: "wasm" });
+    } else if (sourceText.get("text").match(/^\s*</)) {
+      // Use HTML mode for files in which the first non whitespace
+      // character is `<` regardless of extension.
+      this.editor.setMode({ name: "htmlmixed" });
+    } else {
+      this.editor.setMode({ name: "text" });
+    }
+  },
```

Comment:
Now that I think a bit about it more, the util methods before were mostly around codemirror functionality like alignLine middle and less around reading state like source text. 

I'm sensitive to letting a component do to much and Editor, but this makes sense.